test/alignak_test.py

```diff
@@ -131,127 +132,94 @@ class Pluginconf(object):
-        self.sched = scheddaemon.sched
-        scheddaemon.load_modules_manager()
-        # Remember to clean the logs we just created before launching tests
-        self.clear_logs()
-        m = MacroResolver()
-        m.init(self.conf)
-        self.sched.load_conf(self.conf)
-        e = ExternalCommandManager(self.conf, 'applyer')
-        self.sched.external_command = e
-        e.load_scheduler(self.sched)
-        e2 = ExternalCommandManager(self.conf, 'dispatcher')
-        e2.load_arbiter(self)
-        self.external_command_dispatcher = e2
-        self.sched.conf.accept_passive_unknown_check_results = False
-
-        self.sched.schedule()
+        self.schedulers = []
+        self.brokers = []
+        self.arbiter = None
+        self.conf_is_correct = False
+        self.configuration_warnings = []
+        self.configuration_errors = []
+
+        self.arbiter = Arbiter([configuration_file], False, False, False, False,
+                              '/tmp/arbiter.log', 'arbiter-master')
+
+        try:
+            self.arbiter.load_config_file()
+            # If this assertion does not match, then there is a bug in the arbiter :)
+            self.assertTrue(self.arbiter.conf.conf_is_correct)
+            self.conf_is_correct = True
+            self.configuration_warnings = self.arbiter.conf.configuration_warnings
+            self.configuration_errors = self.arbiter.conf.configuration_errors
+        except SystemExit:
+            self.configuration_warnings = self.arbiter.conf.configuration_warnings
+            print("Configuration warnings:")
+            for msg in self.configuration_warnings:
+                print(" - %s" % msg)
+            self.configuration_errors = self.arbiter.conf.configuration_errors
+            print("Configuration errors:")
+            for msg in self.configuration_errors:
+                print(" - %s" % msg)
+            raise
+
+        for broker in self.arbiter.conf.brokers:
+            self.brokers.append(broker)
+
+        for arb in self.arbiter.conf.arbiters:
+            if arb.get_name() == self.arbiter.config_name:
+                self.arbiter.myself = arb
+        self.arbiter.dispatcher = Dispatcher(self.arbiter.conf, self.arbiter.myself)
+        self.arbiter.dispatcher.prepare_dispatch()
+
+        for scheduler in self.arbiter.dispatcher.schedulers:
+            sched = Alignak([], False, False, True, '/tmp/scheduler.log')
+            # logger.setLevel('DEBUG')
+            sched.load_modules_manager()
+            sched.new_conf = scheduler.conf_package
+            if sched.new_conf:
+                sched.setup_new_conf()
+            self.schedulers.append(sched)
 
     def add(self, b):
         if isinstance(b, Brok):
             self.broks[b.uuid] = b
             return
         if isinstance(b, ExternalCommand):
-            self.sched.run_external_command(b.cmd_line)
+            self.schedulers[0].run_external_command(b.cmd_line)
```

Comment:
So we add all broker to the first sched? Maybe we could at least always take the "scheduler-master" or something. So that we ensure we add to the same scheduler. This may required to have schedulers as dict and not lists